libtransmission/net.cc

```diff
@@ -556,6 +560,92 @@ std::pair<tr_address, std::byte const*> tr_address::from_compact_ipv6(std::byte
     return { address, compact };
 }
 
+std::optional<unsigned> tr_address::to_interface_index() const noexcept
+{
+    if (!is_valid())
+    {
+        tr_logAddDebug("Invalid target address to find interface index");
+        return {};
+    }
+
+    tr_logAddDebug(fmt::format("Find interface index for {}", display_name()));
+
+#ifdef _WIN32
+    auto p_addresses = std::unique_ptr<void, void (*)(void*)>{ nullptr, operator delete };
+    for (auto p_addresses_size = ULONG{ 15000 } /* 15KB */;;)
```

Comment:
Could be worth commenting that this value comes from https://learn.microsoft.com/en-us/windows/win32/api/iphlpapi/nf-iphlpapi-getadaptersaddresses

Reply:
5358c7a